Extension/src/LanguageServer/client.ts

```diff
@@ -2180,24 +2203,82 @@ export class DefaultClient implements Client {
     }
 
     /**
-     * returns a promise that waits for the all pendingTasks are complete (e.g. language client is ready for use)
+     * a Promise that can be awaited to know when it's ok to proceed.
+     *
+     * This is a lighter-weight complement to `enqueue()`
+     *
+     * Use `await <client>.ready` when you need to ensure that the client is initialized, and to run in order
+     * Use `enqueue()` when you want to ensure that subsequent calls are blocked until a critical bit of code is run.
+     *
+     * This is lightweight, because if the queue is empty, then the only thing to wait for is the client itself to be initialized
      */
-    get ready(): Promise<void> {
-        return this.isSupported ? Promise.resolve(pendingTask?.getPromise() || undefined) : fail(localize("unsupported.client", "Unsupported client"));
+    get ready():  Promise<unknown> {
+        if(!DefaultClient.dispatching.isCompleted || DefaultClient.queue.length) {
```

Comment:
I'm surprised vscode isn't auto-formatting these `if(`'s for you.  There should be a space between keywords and parentheses.

Reply:
Yeah, I'd prefer `if (`.

Reply:
I also see `while(` below.